db/migrate/20170313192244_create_new_db_schema_with_generic_page_version_change_annotation.rb

```diff
@@ -1,10 +1,11 @@
 class CreateNewDbSchemaWithGenericPageVersionChangeAnnotation < ActiveRecord::Migration[5.0]
   def change
-    # Postgres needs this for proper UUID support
-    enable_extension 'uuid-ossp'
+    # Since 9.4, PostgreSQL recommends using `pgcrypto`'s `gen_random_uuid`
+    # https://github.com/rails/rails/commit/b915b11cca558eb99b7c2621c4457491d4bdb43b
+    enable_extension 'pgcrypto'
 
-    create_table :pages, id: false do |t|
-      t.primary_key :uuid, :uuid
+    create_table :pages, id: false, default: -> { 'gen_random_uuid' } do |t|
+      t.primary_key :uuid, gen_random_uuid
```

Comment:
Hmmm, I don’t think you actually need to make these changes to the `default` option for `create_table` or the type of the primary key.

For me, `schema.rb` gets updated showing that `gen_random_uuid()` is being used, but only if the migrations don’t fail (in which case I think it’s just that `schema.rb` never gets written out to disk). Since migration 20170517222310 is failing, you need to not run that, for example—I tried `rake db:migrate VERSION=20170508221326`.

I think the behavior might also depend on what version of PG you have. I’m running 9.6.